File: TheepotMobile/TheepotMobile/Services/MeldingRouter.swift
import Foundation

/// Gedeelde deeplink-state: AppDelegate zet `gewenstGesprekId` bij een tik op
/// een pushmelding, DashboardView/ChatListView lezen dat om naar het juiste
/// gesprek te navigeren. `actiefGesprekId` houdt bij welk gesprek open staat,
/// zodat AppDelegate de banner kan onderdrukken voor een gesprek dat je al ziet.
@MainActor
final class MeldingRouter: ObservableObject {
    @Published var gewenstGesprekId: String?
    @Published var actiefGesprekId: String?
    /// Zelfde patroon als `gewenstGesprekId`, maar voor een tik op een
    /// agenda-herinnering: AppDelegate zet dit, AgendaView pakt het op.
    @Published var gewenstAfspraakId: String?

    func open(gesprekId: String) {
        gewenstGesprekId = gesprekId
    }

    func verwerkt() {
        gewenstGesprekId = nil
    }

    func open(afspraakId: String) {
        gewenstAfspraakId = afspraakId
    }

    func verwerktAfspraak() {
        gewenstAfspraakId = nil
    }
}

File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/push/MeldingRouter.kt
package nl.bsodetheepot.mobile.data.push

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

/**
 * Gedeelde deeplink-state, analoog aan MeldingRouter.swift. MainActivity zet
 * `gewenstGesprekId` als de app via een pushmelding (of de intent bij cold
 * start) geopend wordt; DashboardScreen/ChatListScreen lezen dat om naar het
 * juiste gesprek te navigeren. `actiefGesprekId` houdt bij welk gesprek open
 * staat; TheepotMessagingService gebruikt dat om `onMessageReceived` de
 * banner te laten onderdrukken (via de notificatie zelf hoeft dat op Android
 * niet apart, want de app tekent zelf de melding — zie TheepotMessagingService).
 */
object MeldingRouter {
    private val _gewenstGesprekId = MutableStateFlow<String?>(null)
    val gewenstGesprekId: StateFlow<String?> = _gewenstGesprekId.asStateFlow()

    private val _actiefGesprekId = MutableStateFlow<String?>(null)
    val actiefGesprekId: StateFlow<String?> = _actiefGesprekId.asStateFlow()

    /** Zelfde patroon als `gewenstGesprekId`, maar voor een tik op een agenda-herinnering. */
    private val _gewenstAfspraakId = MutableStateFlow<String?>(null)
    val gewenstAfspraakId: StateFlow<String?> = _gewenstAfspraakId.asStateFlow()

    fun open(gesprekId: String) {
        _gewenstGesprekId.value = gesprekId
    }

    fun verwerkt() {
        _gewenstGesprekId.value = null
    }

    fun zetActiefGesprek(gesprekId: String?) {
        _actiefGesprekId.value = gesprekId
    }

    fun openAfspraak(afspraakId: String) {
        _gewenstAfspraakId.value = afspraakId
    }

    fun verwerktAfspraak() {
        _gewenstAfspraakId.value = null
    }
}
